StyleCop.Analyzers/StyleCop.Analyzers.Test/Verifiers/DiagnosticVerifier.cs

```diff
@@ -18,17 +17,42 @@ public abstract partial class DiagnosticVerifier
         protected static DiagnosticResult[] EmptyDiagnosticResults { get; } = { };
 
         /// <summary>
-        /// Get the C# analyzer being tested - to be implemented in non-abstract class.
+        /// Get the C# analyzer being tested
         /// </summary>
         /// <returns>
         /// A new instance of the C# analyzer being tested.
         /// </returns>
+        /// <remarks>
+        /// <para>
+        /// This method or <see cref="GetCSharpDiagnosticAnalyzers"/> must be implemented in a non-abstract base class.
+        /// </para>
+        /// </remarks>
         protected virtual DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
             return null;
         }
 
         /// <summary>
+        /// Gets the C# analyzers being tested
+        /// </summary>
+        /// <returns>
+        /// New instances of all the C# analyzers being tested.
+        /// </returns>
+        /// <remarks>
+        /// <para>
+        /// This method or <see cref="GetCSharpDiagnosticAnalyzer"/> must be implemented in a non-abstract base class.
+        /// </para>
+        /// <para>
+        /// The default implementation of this method will call <see cref="GetCSharpDiagnosticAnalyzer"/>.
+        /// Overriding this method will therefore ignore any override of <see cref="GetCSharpDiagnosticAnalyzer"/>.
+        /// </para>
+        /// </remarks>
+        protected virtual DiagnosticAnalyzer[] GetCSharpDiagnosticAnalyzers()
```

Comment:
:bulb: Let's have just one method which returns `IEnumerable<DiagnosticAnalyzer>`. This will allow us to update the existing overloads to just use `yield return` instead of `return`.